index.js

```
@@ -1,43 +1,74 @@
 'use strict';
-const mem = require('mem');
 const mimicFn = require('mimic-fn');
+const mapAgeCleaner = require('map-age-cleaner');
 
-const memoizedFunctions = new WeakMap();
+const cacheStore = new WeakMap();
 
 const pMemoize = (fn, {cachePromiseRejection = false, ...options} = {}) => {
+	const {maxAge, cacheKey} = options;
 	const cache = options.cache || new Map();
-	const cacheKey = options.cacheKey || (([firstArgument]) => firstArgument);
 
-	const memoized = mem(fn, {
-		...options,
-		cache,
-		cacheKey
-	});
+	if (!isNaN(maxAge)) {
```

It should fail loudly with a `TypeError` if `maxAge` doesn't pass `Number.isSafeInteger`.

---

OK. Thanks, this if condition didn't make sense.